### PR TITLE
Update init.example.el with (rust +lsp)

### PR DESCRIPTION
**Pull request by c1ttim**
Today I had tried to setup clean Doom Emacs for rust and got debacle by racer compile errors:

> error: could not compile "racer" due to 3 previous errors
> warning: build failed, waiting for other jobs to finish...
> error: failed to compile "racer v2.2.2"

My software:

- Emacs 28.1
- Doom Core v3.0.0-dev
- Doom Modules v22.08.0-dev
- rustup 1.25.1
- racer v2.2.2 (which wont compiled)
- rust v1.63.0
- rust v1.65.0-nightly

After a quick review of [racer github page](https://github.com/racer-rust/racer), it became clear that racer is no more in active development and racer developers directly sending users to use rust-analyzer or another similar tools.

Summing up my experience with a fresh components which I provided above, it's time to use lsp for rust by default.

Fixes #0000
References #0000
Replaces #0001

-----
- [x] I searched the issue tracker and this hasn't been PRed before.
- [x] My changes are not on [the do-not-PR list](https://doomemacs.org/d/do-not-pr) for this project.
- [x] My commits conform to [the git conventions](https://doomemacs.org/d/git-conventions).

### CODE REVIEW

**Review by hlissner — APPROVED** [review]

**Comment by hlissner** [comment]
I agree that Racer is EOL, so I've altogether removed support for it in 63b0ebe, which incorporates your suggestion, so I'll consider this issue resolved. Thanks for bringing it to my attention!
